### DISCUSSION TITLE
Importing 3rd party packages via npm

### DISCUSSION THREAD

**Discussion by kamesh-git** [vercel/next.js#34548] [Help] [answered]
I'm trying to use a npm package [Markdown to html](https://www.npmjs.com/package/markdown-to-html-converter) in my Next js project but whenever i run the code i'm rendering with this error consistently
`error - SyntaxError: Cannot use import statement outside a module`
could anyone please help me out or atleast give me a insight of the concepts which i want to look for.

Thank you

**Comment by 2stash** [comment]
@kamesh-git Hi, can you share how you are using it? How are you importing this module to this file?

Here is a stackover flow answer that might help if you are importing the module incorrectly.

https://stackoverflow.com/questions/58211880/uncaught-syntaxerror-cannot-use-import-statement-outside-a-module-when-import

**Reply by kamesh-git** [reply]
@2stash thank you for sharing your thoughts but how to use this syntax in Next js because we are working with only .js files and no html files and these are the lines i'm using in my code

`import convertToHTML from 'markdown-to-html-converter`
`const markdownStr = '# Example \nThis is my example text';`
`const htmlStr = convertToHTML(markdownStr);`

and more importantly the package is working very fine in react apps with the same lines

**Reply by 2stash** [reply]
@kamesh-git Hi, yeah you didn't include any info in your first post so had to just guess. 

I started up a new react app and a new next js app with: create-react-app and create-next-app. For both I imported the markdown-to-html-converter. The react app worked as you said, and the next-app did not work with an "error - SyntaxError: Cannot use import statement outside a module".

To solve this problem I used dynamic imports and disabled server side rendering. Below is a simple example that removes the error. Note, I just display the html as a string and didn't bother to actually display it as html, but I get no errors and the markdown is converted.

https://github.com/2stash/nextjs-dynamic-import

Now why you have to use dynamic imports, or if there is another way around this, I have no idea. I'm still learning next.js myself. Good luck!

**Comment by kamesh-git** [comment, marked as answer]
Awesome and thank you @2stash. it's really a nice explanation, now i got a clear idea https://github.com/vercel/next.js/discussions/34548#discussioncomment-2215325
